load_confounds/parser.py

```diff
@@ -332,11 +334,18 @@ def _load_single(self, confounds_raw):
             )
             confounds = pd.concat([confounds, confounds_compcor], axis=1)
 
+
+        # Convert from DataFrame to numpy ndarray
         labels = confounds.columns
+        confounds = confounds.values
+
+        # Derivatives have NaN
+        # Replace them by zeros, otherwise nilearn will not regress them out.
+        confounds[np.isnan(confounds)] = 0
+
+        # Optionally demean confounds
         if self.demean:
             confounds = scale(confounds, axis=0, with_std=False)
```

Comment:
If you demean after replacing nans by zeros, these won't be zero anymore. Might be a problem in regression.

Reply:
`scale(confounds, axis=0, with_std=False,copy=False)` for in-place computation

Reply:
as scaling is robust to nans (https://github.com/scikit-learn/scikit-learn/blob/fd237278e/sklearn/preprocessing/_data.py#L159) , it would be better to run it before:

Reply:
@HanadS noted that we are expecting nans only on the first row of the derivatives, so it may make more sense to duplicate the values of the second row in the first row.

Reply:
It seems to me that it would make more sense to regress with zero values, and not correcting that time point for derivatives rather than guessing that the motion was similar/continuous.

Reply:
ignoring the effect of derivatives in the regression would require to change the way the regression is implemented, in nilearn. Replacing with zero will not exclude the time point (this is a multivariate process), I don't see a work around.

Also even if hypothetically there was a patch submitted to nilearn, it would add a lot of complexity in the code, while the impact of that single time point will likely be very limited.

Reply:
I don't know which type of regression is used afterwards, I thought this would be a GLM. But sure, this will have limited impact.

Reply:
what's implemented in nilearn is an orthogonal projection with a QR decomposition. I have not thought of the impact setting a particual time point to zero there would have. For a traditional GLM I have not really thought about it, there is the inversion of X'X. Putting a coeff to zero in one of the vectors means that this term will be absent in the coefficients of X'X, but you now have some coefficients with, say, N terms while the other ones have N-1. And then you invert that. Not sure what happens there and if the resulting coefficients are somehow equivalent to ignoring the time point.  

Replacing by the closest value in time seems a relatively safe move to me, as the attribution of the gradient `y[t]-y[t-1]` to `gradient[t]` or `gradient[t-1]` is kind of arbitrary anyway.